packages/datadog-instrumentations/src/mongodb-core.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+global.foo = channel('apm:mongodb:insert:start')
+
+addHook({ name: 'mongodb-core', versions: ['2 - 3.1.9'] }, Server => {
+  const serverProto = Server.Server.prototype
+  shimmer.wrap(serverProto, 'command', command => wrapCommand(command, 'command'))
+  shimmer.wrap(serverProto, 'insert', insert => wrapCommand(insert, 'insert', 'insert'))
+  shimmer.wrap(serverProto, 'update', update => wrapCommand(update, 'update', 'update'))
+  shimmer.wrap(serverProto, 'remove', remove => wrapCommand(remove, 'remove', 'remove'))
+
+  const cursorProto = Server.Cursor.prototype
+  shimmer.wrap(cursorProto, '_getmore', _getmore => wrapCursor(_getmore, 'getMore', 'getMore'))
+  shimmer.wrap(cursorProto, '_find', _find => wrapQuery(_find, '_find'))
+  shimmer.wrap(cursorProto, 'kill', kill => wrapCursor(kill, 'killCursors', 'killCursors'))
+  return Server
+})
+
+addHook({ name: 'mongodb', versions: ['>=4'], file: 'lib/cmap/connection.js' }, Connection => {
+  const proto = Connection.Connection.prototype
+  shimmer.wrap(proto, 'command', command => wrapConnectionCommand(command, 'command'))
+  shimmer.wrap(proto, 'query', query => wrapConnectionCommand(query, 'query'))
+  shimmer.wrap(proto, 'getMore', getMore => wrapConnectionCommand(getMore, 'getMore', 'getMore'))
+  shimmer.wrap(proto, 'killCursors', killCursors => wrapConnectionCommand(killCursors, 'killCursors', 'killCursors'))
+  return Connection
+})
+
+addHook({ name: 'mongodb', versions: ['>=3.3 <4'], file: 'lib/core/wireprotocol/index.js' }, wp => wrapWp(wp))
```

Comment:
Should the instrumentation for `mongodb` be separated now that the instrumentations are decoupled from the plugins?

Reply:
@bengl ?

Reply:
I'm not sure what's being asked here.

Reply:
I was just wondering if the instrumentation should be split in 2 files, one for `mongodb-core` and one for `mongodb` but since they share most of the code maybe that would just make things more complicated.